storage/entities/peer.js

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const { defaults, omit, pick } = require('lodash');
+const filterType = require('../utils/filter_types');
+const { stringToByte } = require('../utils/inputSerializers');
+const BaseEntity = require('./base_entity');
+
+const defaultCreateValues = {};
+
+const readOnlyFields = [];
+
+/**
+ * Peer
+ * @typedef {Object} Peer
+ * @property {number} id
+ * @property {string} ip
+ * @property {number} wsPort
+ * @property {number} state
+ * @property {string} os
+ * @property {string} version
+ * @property {number} clock
+ * @property {string} broadhash
+ * @property {number} height
+ */
+
+/**
+ * Peer Filters
+ * @typedef {Object} filters.Peer
+ * @property {number} [id]
+ * @property {number} [id_eql]
+ * @property {number} [id_ne]
+ * @property {number} [id_gt]
+ * @property {number} [id_gte]
+ * @property {number} [id_lt]
+ * @property {number} [id_lte]
+ * @property {number} [id_in]
```

Comment:
I believe we don't need filter on `id`, its just for internal use. Logically `ip` should the primary key and no duplicate peer should exist in the table.